sdks/python/apache_beam/io/gcp/bigtableio.py

```diff
@@ -141,3 +148,127 @@ def expand(self, pvalue):
             | beam.ParDo(_BigTableWriteFn(beam_options['project_id'],
                                           beam_options['instance_id'],
                                           beam_options['table_id'])))
+
+
+class BigtableSource(iobase.BoundedSource):
+  def __init__(self, project_id, instance_id, table_id, filter_=None):
+    """ Constructor of the Read connector of Bigtable
+
+    Args:
+      project_id: [string] GCP Project of to write the Rows
+      instance_id: [string] GCP Instance to write the Rows
+      table_id: [string] GCP Table to write the `DirectRows`
+      filter_: [RowFilter] Filter to apply to columns in a row.
+    """
+    super(self.__class__, self).__init__()
+    self._init({'project_id': project_id,
+                'instance_id': instance_id,
+                'table_id': table_id,
+                'filter_': filter_})
+
+  def __getstate__(self):
+    return self.beam_options
+
+  def __setstate__(self, options):
+    self._init(options)
+
+  def _init(self, options):
+    self.beam_options = options
+    self.table = None
+    self.sample_row_keys = None
+    self.row_count = Metrics.counter(self.__class__.__name__, 'Row count')
+
+
+  def _get_table(self):
+    if self.table is None:
+      self.table = Client(project=self.beam_options['project_id'])\
+                      .instance(self.beam_options['instance_id'])\
+                      .table(self.beam_options['table_id'])
+    return self.table
+
+  def get_sample_row_keys(self):
+    """ Get a sample of row keys in the table.
+
+    The returned row keys will delimit contiguous sections of the table of
+    approximately equal size, which can be used to break up the data for
+    distributed tasks like mapreduces.
+    :returns: A cancel-able iterator. Can be consumed by calling ``next()``
+    			  or by casting to a :class:`list` and can be cancelled by
+    			  calling ``cancel()``.
+    """
```

Comment:
Due to some unusual behavior, added the following comment here:
`    ***** NOTE: For unclear reasons, the function returns generator even
    after wrapping the result as a list. In order to be used as a list, the
    result should be wrapped as a list AGAIN! E.g., see 'estimate_size()'`